src/object-specs.ts

```diff
@@ -49,33 +49,49 @@ const BASE_PLUGINS:List.List<string> = List.of(
   'use-cpp'
 );
 
-interface ValueObjectCreationContext {
+interface ObjectSpecCreationContext {
   baseClassName:string;
   baseClassLibraryName:Maybe.Maybe<string>;
   diagnosticIgnores:List.List<string>;
-  plugins:List.List<ValueObject.Plugin>;
+  plugins:List.List<ObjectSpec.Plugin>;
   defaultIncludes:List.List<string>;
 }
 
+type ObjectSpecExtension =
+    "value" |
+    "object";
+
 interface PathAndTypeInfo {
   path: File.AbsoluteFilePath;
-  typeInformation: ValueObject.Type;
+  typeInformation: ObjectSpec.Type;
+}
+
+function modifyFoundTypeBasedOnExtension(foundType:ObjectSpec.Type, extension:ObjectSpecExtension):ObjectSpec.Type {
+  switch (extension) {
+    case "value":
+      foundType.includes = foundType.includes.concat(ObjectSpec.VALUE_OBJECT_SEMANTICS);
```

Comment:
Don't mutable the foundType, create a new type. I think you should be able to use the spread operator to make that easier